objects/src/transaction/proven_tx.rs

```diff
@@ -52,10 +48,7 @@ pub struct ProvenTransaction {
     input_notes: InputNotes<Nullifier>,
 
     /// The id and  metadata of all notes created by the transaction.
-    output_notes: OutputNotes<NoteEnvelope>,
-
-    /// Optionally the output note's data, used to share the note with the network.
-    output_note_details: BTreeMap<NoteId, Note>,
```

Comment:
This is no longer necessary, the enum gives better type safety since the data is not split into two places.